project-api-tests/src/test/java/org/ballerina/projectapi/BalToolTest.java

```diff
@@ -0,0 +1,1008 @@
+import static org.ballerina.projectapi.CentralTestUtils.getEnvVariables;
+import static org.ballerina.projectapi.CentralTestUtils.getString;
+import static org.ballerina.projectapi.CentralTestUtils.isToolAvailableInCentral;
+import static org.ballerina.projectapi.TestUtils.DISTRIBUTION_FILE_NAME;
+import static org.ballerina.projectapi.TestUtils.OUTPUT_CONTAIN_ERRORS;
+import static org.ballerina.projectapi.TestUtils.executeCommand;
+import static org.ballerina.projectapi.TestUtils.executeHelpCommand;
+import static org.ballerina.projectapi.TestUtils.executeToolCommand;
+
+public class BalToolTest {
+    private Path tempWorkspaceDirectory;
+    private final String orgName = "bctestorg";
+    private static final String toolId = "disttest";
+    private static final String nonExistingTool = "disttest2";
+    private static final String latestVersion = "1.1.0";
+    private static final String specificVersion = "1.0.0";
+    private static final String nonExistingVersion = "1.0.3";
+    private static final String incompatibleDistVersion = "1.0.4";
+    private static final String packageName = "disttest";
+    private Map<ToolSubCommand, ToolEnvironment> toolEnvironments;
+
+    @BeforeClass()
+    public void setUp() throws IOException, InterruptedException {
+        TestUtils.setupDistributions();
+        tempWorkspaceDirectory = Files.createTempDirectory("bal-test-integration-packaging-workspace-");
+        setToolEnvironmentsForSubCommands();
+        Map<String, String> envVariables = getEnvVariables();
+        envVariables.put(BALLERINA_DEV_CENTRAL, "true");
+        if (!isToolAvailableInCentral(toolId, tempWorkspaceDirectory, envVariables)) {
+            Assert.fail("Tool " + toolId + " is not available in central");
+        }
+    }
+
+    @BeforeGroups(value = "pull")
+    public void setupPullTests() {
+        // Remove the tool from bal-tools.toml if exists
+        BalToolsToml balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.PULL).balToolsTomlPath);
+        BalToolsManifest balToolsManifest = BalToolsManifestBuilder.from(balToolsToml).build();
+        balToolsManifest.removeTool(toolId);
+        balToolsToml.modify(balToolsManifest);
+
+        // Delete the tool from local cache if exists
+        Path toolPath = toolEnvironments.get(ToolSubCommand.PULL).centralCachePath
+                .resolve(Path.of(orgName, packageName));
+        if (Files.exists(toolPath)) {
+            ProjectUtils.deleteDirectory(toolPath);
+        }
+    }
+
+    @Test(description = "Pull tool without specifying a version", groups = {"pull"})
+    public void testPullToolWithoutVersion() throws IOException, InterruptedException {
+        Process cmdExec = executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("pull", toolId)), toolEnvironments.get(ToolSubCommand.PULL).envVariables);
```

Comment:
```suggestion
                new ArrayList<>(Arrays.asList("PULL", toolId)), toolEnvironments.get(ToolSubCommand.PULL).envVariables);
```
shall we use a constant for this as this is used in multiple places?

Reply:
Fixed with 4655a12b0cf589483ba556c28259e22d5beb4126